lgc/test/ElfRelocationSize.lgc

```diff
@@ -0,0 +1,221 @@
+; RUN: lgc -mcpu=gfx1030 -extract=2 -o %t.fs.elf %s
+; RUN: lgc -mcpu=gfx1030 -extract=3 -other=%t.fs.elf -o %t.vs.elf %s
+; RUN: lgc -mcpu=gfx1030 -extract=1 -l %s -o %t.pipe.elf %t.vs.elf %t.fs.elf
+; RUN: readelf -W -S %t.pipe.elf | grep -P -o ".rel.text\s+REL\s+[0-9]+\s+\S+" | awk '{ print $4 }' >%t.offs1
+; RUN: readelf -W -S %t.pipe.elf | grep -P -o ".rodata.cst32\s+PROGBITS\s+[0-9]+\s+\S+" | awk '{ print $4 }' >%t.offs2
+; RUN: echo "obase=16;ibase=16;$(cat %t.offs2 | tr a-z A-Z)-$(cat %t.offs1 | tr a-z A-Z)" | bc | FileCheck %s
+
+; This test checks that no random extra bytes are generated after the relocations
+; in the relocation section (.rel.text). To check that, we extract the offsets of
+; the relocation section and the following section from the elf and subtract
+; them. Here we write 4 16-bit relocations, so the printed value should be
+; around 0x40 (because of section alignment).
+
+; CHECK: 41
```

Comment:
```suggestion
; This test checks that no random extra bytes are generated after the relocations
; in the relocation section (`.rel.text`). To check that, we extract the offsets of
; the relocation section and the following section from the elf and subtract
; them. Here we write 4 16-bit relocations, so the printed value should be
; around 0x40 (65 dec) (because of section alignment).

; RUN: lgc -mcpu=gfx1030 -extract=2 -o %t.fs.elf %s
; RUN: lgc -mcpu=gfx1030 -extract=3 -other=%t.fs.elf -o %t.vs.elf %s
; RUN: lgc -mcpu=gfx1030 -extract=1 -l %s -o %t.pipe.elf %t.vs.elf %t.fs.elf
;
; RUN: llvm-readelf %t.pipe.elf --section-headers --elf-output-style=LLVM | FileCheck %s --match-full-lines
;
; CHECK-LABEL: Name: .rel.text ({{[0-9]+}})
; CHECK:       Offset: 0x[[#%X,OFFSET1:]]
; CHECK-LABEL: Name: .rodata.cst32 ({{[0-9]+}})
; CHECK:       Offset: 0x[[#OFFSET1 + 65]]
; CHECK-LABEL: Name: .note.GNU-stack ({{[0-9]+}})
```